crates/types/src/rpc/transactions.rs

```diff
@@ -272,3 +278,179 @@ pub enum BroadcastedDeclareTransaction {
     V1(Box<BroadcastedDeclareTransactionV1>),
     V2(Box<BroadcastedDeclareTransactionV2>),
 }
+
+/// Flags that indicate how to simulate a given transaction.
+/// By default, the sequencer behavior is replicated locally (enough funds are expected to be in the
+/// account, and fee will be deducted from the balance before the simulation of the next
+/// transaction). To skip the fee charge, use the SKIP_FEE_CHARGE flag.
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize)]
+pub enum SimulationFlag {
+    #[serde(rename = "SKIP_VALIDATE")]
+    SkipValidate,
+    #[serde(rename = "SKIP_FEE_CHARGE")]
+    SkipFeeCharge,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub enum CallType {
+    #[serde(rename = "LIBRARY_CALL")]
+    LibraryCall,
+    #[serde(rename = "CALL")]
+    Call,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct FunctionInvocation {
+    #[serde(flatten)]
+    function_call: FunctionCall,
+    caller_address: Felt,
+    class_hash: Felt,
+    entry_point_type: EntryPointType,
+    call_type: CallType,
+    result: Vec<Felt>,
+    calls: Vec<FunctionInvocation>,
+    events: Vec<Event>,
+    messages: Vec<MessageToL1>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(untagged)]
+pub enum TransactionTrace {
+    Invoke(InvokeTransactionTrace),
+    Declare(DeclareTransactionTrace),
+    DeployAccount(DeployAccountTransactionTrace),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Reversion {
+    pub revert_reason: String,
+}
+
+#[derive(Debug, Clone, Serialize)]
+#[serde(untagged)]
+pub enum ExecutionInvocation {
+    Succeeded(FunctionInvocation),
+    Reverted(Reversion),
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct InvokeTransactionTrace {
+    pub validate_invocation: Option<FunctionInvocation>,
+    pub execution_invocation: ExecutionInvocation,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct DeclareTransactionTrace {
+    pub validate_invocation: Option<FunctionInvocation>,
+    pub fee_transfer_invocation: Option<FunctionInvocation>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct DeployAccountTransactionTrace {
+    pub validate_invocation: Option<FunctionInvocation>,
+    pub constructor_invocation: Option<FunctionInvocation>,
+    pub fee_transfer_invocation: Option<FunctionInvocation>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct SimulatedTransaction {
+    pub transaction_trace: TransactionTrace,
+    pub fee_estimation: FeeEstimateWrapper,
+}
+
+impl TryFrom<L2toL1MessageInfo> for MessageToL1 {
+    type Error = Error;
+
+    fn try_from(value: L2toL1MessageInfo) -> Result<Self, Self::Error> {
+        Ok(Self {
+            from_address: value.from_address.try_into()?,
+            // to_address: starknet_api::transaction::EthAddress(value.to_address.0.into()),
```

Comment:
commented unused code

Reply:
Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/189/commits/adccc8d8bcf6a11acfc38fc29a2637c4a7319a57